### PR TITLE
CI: Add OpenBSD build tests for staged commits

### PR DESCRIPTION
**Pull request by kinkie**
Use a GitHub-hosted VM to create OpenBSD test environment. Requires
GitHub repository configuration that permits the use of
`vmactions/openbsd-vm@v1` Action.

We have not enabled ccache optimization for OpenBSD tests because we
do not know how to copy updated ccache files from VM back into runner.

### CODE REVIEW

File: .github/workflows/slow.yaml [hunk 1]
```diff
@@ -181,3 +181,61 @@ jobs:
...
+              libltdl \
+              libtool \
+              libnettle \
+              libxml \
+              libtool \
+              m4 \
+              metaauto
+
+            mkdir $HOME/bin
+            ln -s /usr/local/bin/ggrep $HOME/bin/grep
```
Comment (kinkie) [inline .github/workflows/slow.yaml:217]: There are a bunch of GNUisms in our build chain

File: .github/workflows/slow.yaml [hunk 2]
```diff
@@ -181,3 +181,59 @@ jobs:
...
+
+          run: |
+            export MAKE=gmake
+            export pjobs="-j`gnproc`"
+            export AUTOMAKE_VERSION=1.16
```
Comment (kinkie) [inline .github/workflows/slow.yaml:222]: OpenBSD support for multi-version automake is complicated

File: .github/workflows/slow.yaml [hunk 3]
```diff
@@ -181,3 +181,59 @@ jobs:
...
+            export amver=${AUTOMAKE_VERSION}
+            export ACLOCAL_AUTOMAKE_DIR="/usr/local/share/aclocal-${AUTOMAKE_VERSION}"
+            export ACLOCAL_PATH="/usr/local/share/aclocal:/usr/local/share/aclocal-${AUTOMAKE_VERSION}"
+            export AUTOCONF_VERSION=2.72
```
Comment (kinkie) [inline .github/workflows/slow.yaml:226]: autoconf, too

File: .github/workflows/slow.yaml [hunk 4]
```diff
@@ -181,3 +181,59 @@ jobs:
...
+            export acver=${AUTOCONF_VERSION}
+            export ltver=2.4.2
+            export CFLAGS='-Wno-compound-token-split-by-macro'
+            export LDFLAGS="-L/usr/local/lib"
+            export PATH="$HOME/bin:$PATH"
```
Comment (kinkie) [inline .github/workflows/slow.yaml:231]: we need this to shadow the system 'grep' command

File: .github/workflows/slow.yaml [hunk 5]
```diff
@@ -181,3 +181,59 @@ jobs:
...
+
+            mkdir $HOME/bin
+            ln -s /usr/local/bin/ggrep $HOME/bin/grep
+
+          run: |
+            export MAKE=gmake
+            export pjobs="-j`gnproc`"
+            export AUTOMAKE_VERSION=1.16
+            export amver=${AUTOMAKE_VERSION}
+            export ACLOCAL_AUTOMAKE_DIR="/usr/local/share/aclocal-${AUTOMAKE_VERSION}"
```
Comment (rousskov) [inline .github/workflows/slow.yaml:224]: Is there a similar hack that can avoid the following [warnings](https://github.com/kinkie/squid/actions/runs/12354857472/job/34477146806)?

```
  aclocal-1.16: warning: couldn't open directory '/usr/bin/../share/aclocal': No such file or directory
```

File: .github/workflows/slow.yaml (hunk 4 above)
Comment (rousskov) [inline .github/workflows/slow.yaml:231]: > we need this to shadow the system 'grep' command

I recommend making that a source comment. Unlike "its complicated" comments elsewhere in this PR, comments about grep-related problems are actually useful information that should be preserved in sources IMHO:

```suggestion
            # until we remove GNUisms from our grep commands,
            # shadow system `grep` with `ggrep` installed/linked earlier
            export PATH="$HOME/bin:$PATH"
```

or

```suggestion
            export PATH="$HOME/bin:$PATH" # shadow system `grep`
```

or something like that.

File: .github/workflows/slow.yaml [hunk 6]
```diff
@@ -181,3 +181,59 @@ jobs:
...
+            export CFLAGS='-Wno-compound-token-split-by-macro'
+            export LDFLAGS="-L/usr/local/lib"
+            export PATH="$HOME/bin:$PATH"
+            ./test-builds.sh
+
+      - name: Publish build logs
+        if: success() || failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: openbsd-build-logs
```
Comment (rousskov) [inline .github/workflows/slow.yaml:238]: To make this name consistent with _all_ existing "Publish build logs" steps:

```suggestion
          name: build-logs-openbsd
```

This is the only change I insist on.

**Review by rousskov — CHANGES_REQUESTED** [review]
Thank you for moving these tests to GitHub. I have only one blocking change request.

File: .github/workflows/slow.yaml (hunk 5 above)
Comment (kinkie) [reply .github/workflows/slow.yaml:224]: Not that I know of. OpenBSD is rough around the edges in these areas

**Review by rousskov — APPROVED** [review]

**Comment by rousskov** [comment]
GitHub [reported](https://github.com/squid-cache/squid/actions/runs/12400873047) "startup failure" for the staged commit [c7e0a60](https://github.com/squid-cache/squid/commit/c7e0a60c59a2ef3ff9564f12ab9d9a11c6cba6ad). AFAICT, GitHub provided no failure details and probably did not report that odd failure to agents monitoring repository state (e.g., Anubis). Thus, this PR got stuck waiting for staging tests that are not running.

I would expect a staged test to fail because our GitHub repository configuration did not allow `vmactions/openbsd-vm@v1` Action. I fixed that and touched PR description to prompt Anubis to stage a new staged commit. That new commit is [75c97d2](https://github.com/squid-cache/squid/commit/75c97d271047f1e6041d49a9a96e3aca2de4cc60). It is being tested now.

**Pull request closed without merging**